src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,279 @@
+
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing
+{
+    /// <summary>
+    /// The ComWrappers implementation for System.Drawing.Common's COM interop usages.
+    ///
+    /// Supports IStream and IPicture COM interfaces.
+    /// </summary>
+    internal unsafe class DrawingComWrappers : ComWrappers
+    {
+        private static readonly ComInterfaceEntry* s_wrapperEntry = InitializeComInterfaceEntry();
+        private static readonly Lazy<DrawingComWrappers> s_instance = new Lazy<DrawingComWrappers>(() => new DrawingComWrappers());
+
+        private DrawingComWrappers() { }
+
+        internal static DrawingComWrappers Instance => s_instance.Value;
+
+        internal static void CheckStatus(int result)
+        {
+            if (result != 0)
+            {
+                throw new ExternalException() { HResult = result };
+            }
+        }
+
+        private static ComInterfaceEntry* InitializeComInterfaceEntry()
+        {
+            GetIUnknownImpl(out IntPtr fpQueryInteface, out IntPtr fpAddRef, out IntPtr fpRelease);
+
+            IStreamVtbl iStreamVtbl = IStreamVtbl.Create(fpQueryInteface, fpAddRef, fpRelease);
+
+            IntPtr iStreamVtblRaw = RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IStreamVtbl), sizeof(IStreamVtbl));
+            Marshal.StructureToPtr(iStreamVtbl, iStreamVtblRaw, false);
+
+            ComInterfaceEntry* wrapperEntry = (ComInterfaceEntry*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IStreamVtbl), sizeof(ComInterfaceEntry));
+            wrapperEntry->IID = Interop.Ole32.IStreamComWrapper.Guid;
+            wrapperEntry->Vtable = iStreamVtblRaw;
+            return wrapperEntry;
+        }
+
+        protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count)
+        {
+            Debug.Assert(obj is Interop.Ole32.IStreamComWrapper);
+            Debug.Assert(s_wrapperEntry != null);
+
+            // Always return the same table mappings.
+            count = 1;
+            return s_wrapperEntry;
+        }
+
+        protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+        {
+            Debug.Assert(flags == CreateObjectFlags.None);
+
+            Guid pictureGuid = IPicture.Guid;
+            int hr = Marshal.QueryInterface(externalComObject, ref pictureGuid, out IntPtr comObject);
+            if (hr == 0)
+            {
+                return new PictureWrapper(comObject);
+            }
+
+            throw new NotImplementedException();
+        }
+
+        protected override void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal struct IUnknownVtbl
+        {
+            public IntPtr QueryInterface;
```

Comment:
It would be more work, but might be easier to understand/more efficient if function pointers and `UnmanagedCallersOnly` was used, right?

Reply:
@AaronRobinsonMSFT - any reason why https://github.com/dotnet/samples/tree/main/core/interop/comwrappers/IDispatch doesn't use function pointers and `UnmanagedCallersOnly`?

Reply:
Sort of. When I wrote the example those features weren't in an available preview. Also, it was a functional example not the most optimal. I don't know how to frame simple quick usage with a highly optimized use case. We can update it with best practices but it is going to be some hairy C#.

Reply:
We should fix the example to use the function pointers. I do not think that it will result in particularly hairy C#.

Reply:
Update https://github.com/dotnet/samples/pull/4673.